### ISSUE TITLE
Collect Macs leaves `ping -b x.x.x.x` (does not kill )

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#167]
**Reported by anonymous on 8 Feb 2006 21:17 UTC**
```
When running "collect macs" the ping -b that is
initiated is not killed off properly. I found this
after monitoring our network traffic, and seeing a
staggering amount of ICMP. After digging further with
lsof, ps, and with some help from the guys in
#oscar-cluster, I found that this process was generated
from the "collect macs" function of the oscar wizard,
or by a SIS script invoked by the wizard. 
```

**Comment by dikim33** [comment]
**Comment by bli on 25 Jun 2006 07:07 UTC**
The code in question is browser:trunk/lib/OSCAR/MAC.pm.

The ```ping -b``` command was executed via a ```system()``` call within a forked process.  The PID that is captured is that of the child process however because a system call is executed, the real PID of the ping command is actually PID+1.

This can be fixed by changing the ```system``` call to an ```exec``` call and also added ```wait``` so that the exit status of the child process is reaped.  This allows the correct PID to be killed.

**Comment by dikim33** [comment]
**Comment by bli on 25 Jun 2006 22:32 UTC**
(In [5047]) Fixes #169 - now the "ping" process is properly killed.
Tested on SUSE Linux 10.0 and Fedora Core 4 x86.
